tests/Backend/Mixed/Workspaces/BaseWorkSpacesTestCase.php

```diff
@@ -0,0 +1,137 @@
+<?php
+
+namespace Keboola\Test\Backend\Mixed\Workspaces;
+
+use Keboola\Csv\CsvFile;
+use Keboola\StorageApi\ClientException;
+use Keboola\StorageApi\Workspaces;
+use Keboola\Test\Backend\Workspaces\Backend\WorkspaceBackendFactory;
+use Keboola\Test\Backend\Workspaces\WorkspacesTestCase;
+
+abstract class BaseWorkSpacesTestCase extends WorkspacesTestCase
+{
+    abstract public function workspaceBackendData();
+
+    abstract public function workspaceMixedAndSameBackendDataWithDataTypes();
+
+    abstract public function workspaceMixedAndSameBackendData();
+
+    /**
+     * @dataProvider  workspaceBackendData
+     * @param $backend
+     * @param $dataTypesDefinition
+     */
+    public function testCreateWorkspaceParam($backend, $dataTypesDefinition)
+    {
+        $workspaces = new Workspaces($this->_client);
+
+        $workspace = $workspaces->createWorkspace([
+            'backend' => $backend,
+        ]);
+        $this->assertEquals($backend, $workspace['connection']['backend']);
+
+        $backend = WorkspaceBackendFactory::createWorkspaceBackend($workspace);
+
+        $backend->createTable("mytable", $dataTypesDefinition);
+    }
+
+    /**
+     * @dataProvider workspaceMixedAndSameBackendDataWithDataTypes
+     * @param $workspaceBackend
+     * @param $sourceBackend
+     * @param $columnsDefinition
+     */
+    public function testLoadUserError($workspaceBackend, $sourceBackend, $columnsDefinition)
+    {
+        if ($this->_client->bucketExists("in.c-mixed-test-" . $sourceBackend)) {
+            $this->_client->dropBucket("in.c-mixed-test-{$sourceBackend}", [
+                'force' => true,
+            ]);
+        }
+        $bucketId = $this->_client->createBucket("mixed-test-{$sourceBackend}", "in", "", $sourceBackend);
+        $sourceTableId = $this->_client->createTable(
+            $bucketId,
+            'transactions',
+            new CsvFile(__DIR__ . '/../../../_data/transactions.csv')
+        );
+
+        $workspaces = new Workspaces($this->_client);
+
+        $workspace = $workspaces->createWorkspace([
+            'backend' => $workspaceBackend,
+        ]);
+
+        $options = [
+            "input" => [
+                [
+                    "source" => $sourceTableId,
+                    "destination" => "transactions",
+                    "columns" => $columnsDefinition,
+                ],
+            ],
+        ];
+
+        // exception should be thrown, as quantity has empty value '' and snflk will complain.
+        try {
+            $workspaces->loadWorkspaceData($workspace['id'], $options);
+        } catch (ClientException $e) {
+            $this->assertEquals($e->getStringCode(), "workspace.tableLoad");
```

Comment:
assertoval bych toho víc - i message, nejen string code...

Reply:
Jakože jestli v message je něco užitečného (tj. co je problém) nebo jen "failed".

Reply:
Původní testy já sem to enom refaktoroval.

Reply:
OK